modules/charon-core/src/main/java/org/wso2/charon3/core/protocol/endpoints/GroupResourceManager.java

```diff
@@ -841,10 +843,14 @@ private void processGroupPatchOperations(Map<String, List<PatchOperation>> patch
             if (patchOperation.getPath() != null) {
                 if (patchOperation.getPath().equals(SCIMConstants.GroupSchemaConstants.DISPLAY_NAME) &&
                         patchOperation.getValues() != null) {
-                    JSONObject valuesPropertyJson = (JSONObject) patchOperation.getValues();
                     JSONObject attributePrefixedJson = new JSONObject();
-
-                    attributePrefixedJson.put(SCIMConstants.GroupSchemaConstants.DISPLAY_NAME, valuesPropertyJson);
+                    if (patchOperation.getValues() instanceof String) {
+                        attributePrefixedJson.put(SCIMConstants.GroupSchemaConstants.DISPLAY_NAME,
+                                patchOperation.getValues());
+                    } else {
```

Comment:
Since this block is the default flow we can extract it out from the else block.

Reply:
The value assigned to attributePrefixedJson is different according to the if condition. If the flow doesn't fall inside the if condition then only the value will be a json object.

Reply:
Ack